app/scripts/components/common/blocks/chart/utils.js

```diff
@@ -46,27 +47,61 @@ export const chartTheme = {
   }
 };
 
-export const getColors = function (steps) {
+function getInterpoateFunction(colorScheme) {
+  const fnName = `interpolate${
+    colorScheme[0].toUpperCase() + colorScheme.slice(1)
+  }`;
+  return d3ScaleChromatic[fnName];
+}
+
+export const getColors = function ({ steps, colorScheme }) {
+  const colorFn = getInterpoateFunction(colorScheme);
   return new Array(steps).fill(0).map((e, idx) => {
-    return interpolateViridis(idx / steps);
+    return colorFn(idx / steps);
   });
 };
 
-export const getBottomAxis = function (dateFormat, isLargeScreen) {
+export const getBottomAxis = function ({
+  dateFormat,
+  isLargeScreen,
+  xAxisLabel
+}) {
   // nivo's limit for ticknum:  https://nivo.rocks/guides/axes/
   const tickNum = isLargeScreen ? 8 : 3;
   return {
+    legend: xAxisLabel,
+    legendOffset: 30,
+    legendPosition: 'end',
     tickValues: tickNum,
     tickSize: 5,
     tickPadding: 5,
     format: dateFormat
   };
 };
 
+export const getLeftAxis = function (yAxisLabel) {
+  return {
+    legend: yAxisLabel,
+    legendOffset: -(chartMargin.left * 0.9),
+    legendPosition: 'end'
+  };
+};
+
 export const getFormattedData = function ({ data, idKey, xKey, yKey }) {
   let minY = 0;
   let maxY = 0;
-  const dataWId = data.reduce((acc, curr) => {
+
+  let dataToSort = [...data]
+    .sort((a, b) => {
+      // sort by alphabetical order (y axis)
+      return a[yKey].toLowerCase().localeCompare(b[yKey].toLowerCase());
+    })
+    .sort((a, b) => {
+      // sort by date (x axis)
+      return new Date(a[xKey]) - new Date(b[xKey]);
+    });
```

Comment:
The date sorting needs to be prevailed. if not, the line will be zig-zagged because of not sorted dates. example screenshot attached. 
![Screen Shot 2022-07-06 at 10 37 25 AM](https://user-images.githubusercontent.com/4583806/177576323-30fbdbae-59ec-49f0-8794-3a9e781b816f.png)

Reply:
To solve for the performance problem you can memoize the calculation.
I agree with keeping the date sort, but both operations have to be merged under one sort function.

Something like `array.sort().sort()` will sort the array twice, and only the last sort will have effect. I assume that what you want to do is sort by the `yKey` and only when two `ykey` are the same, then sort by `xkey`.

Assume ascending sort for this example:
```
[
  {ykey: 'a', xKey: 3},
  {ykey: 'a', xKey: 1},
  {ykey: 'b', xKey: 2}
]
```
I imaging that the result you want is sort by `yKey` and then `xKey` so:
```
[
  {ykey: 'a', xKey: 1},
  {ykey: 'a', xKey: 3},
  {ykey: 'b', xKey: 2}
]
```
However with the code above the result would be:
```
[
  {ykey: 'a', xKey: 1},
  {ykey: 'b', xKey: 2},
  {ykey: 'a', xKey: 3}
]
```

Or am I getting this wrong?

Reply:
Sorry to barge in, but just want to also suggest avoiding using `toLowerCase` if you're using `localeCompare`. When using `localeCompare`, consider using relevant `sensitivity` values for the available `options`.

See:
- <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare>
- <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator#options>

Reply:
Interesting. Thanks @chuckwondo

Reply:
@chuckwondo wow you came out of nowhere 😄 thanks! 

@danielfdsilva Yes the result would be like you said. because each unique yKey will get grouped together, when we pass it to Nivo,  intended result is the data like below

```
[{
  yKey: 'b', xKey: 1,
  yKey: 'a', xKey: 1,
  yKey: 'b', xKey: 0,
  yKey: 'a', xKey: 0,
}]
```
gets sorted into 
```
[{
  yKey: 'a', xKey: 0,
  yKey: 'b', xKey: 0,
  yKey: 'a', xKey: 1,
  yKey: 'b', xKey: 1,
}]
```
the data like above.